usage/rest-server/src/main/java/brooklyn/rest/security/provider/DelegatingSecurityProvider.java

```diff
@@ -94,10 +97,16 @@ private synchronized SecurityProvider loadDelegate() {
                 delegate = constructor.newInstance(mgmt);
             } catch (Exception e) {
                 constructor = clazz.getConstructor();
-                delegate = constructor.newInstance();
+                Object delegateO = constructor.newInstance();
+                if (!(delegateO instanceof SecurityProvider)) {
+                    // if classloaders get mangled it will be a different CL's SecurityProvider
+                    throw new ClassCastException("Delegate is either not a security provider or has an incompatible classloader: "+delegateO);
```

Comment:
The reason for the different class loaders is that `BrooklynApiRestClientTest` [configures the `WebAppContext` with `.setExtraClasspath("./target/test-rest-server/");`](https://github.com/apache/incubator-brooklyn/blob/317b69e17338bb541bc903b5ed97e30e58a929a6/usage/rest-client/src/test/java/brooklyn/rest/client/BrooklynApiRestClientTest.java#L70). The additional path contains the server classes, but not the test classes. This causes the test classes to be loaded through the app class loader, while the web app class loader sees the non-test classes.
Why the extra classpath? If there's not a particular reason, then better let the app classloader handle the server classes. This will also help when changing them and running the tests in the client project in Eclipse - currently both projects need to be rebuild for the changes to take effect.
If there's a good reason to keep the extra classpath, then in pom.xml install the `<classifier>tests</classifier>` of `brooklyn-rest-server`.